app/XlsImporter/Foundation/XlsValidator/Traits/ErrorValidationRules.php

```diff
@@ -0,0 +1,470 @@
+     * @return array
+     */
+    public function errorForRecipientCountry(array $activity): array
+    {
+        return (new RecipientCountryRequest())->getErrorsForRecipientCountry(Arr::get($activity, 'recipient_country', []), true, );
+    }
+
+    /**
+     * returns rules for recipient region.
+     * @param array $activity
+     * @return array
+     */
+    public function errorForRecipientRegion(array $activity): array
+    {
+        return (new RecipientRegionRequest())->getErrorsForRecipientRegion(Arr::get($activity, 'recipient_region', []), true, Arr::get($activity, 'recipient_country', []));
+    }
+
+    /**
+     * returns rules for sector form.
+     * @param array $activity
+     * @return array
+     */
+    protected function errorForSector(array $activity): array
+    {
+        return (new SectorRequest())->getErrorsForSector(Arr::get($activity, 'sector', []), true);
+    }
+
+    /**
+     * returns rules for location form.
+     * @param array $activity
+     * @return array
+     */
+    protected function errorForLocation(array $activity): array
+    {
+        return (new LocationRequest())->getErrorsForLocation(Arr::get($activity, 'location', []));
+    }
+
+    /**
+     * returns rules for country budget item form.
+     * @param array $activity
+     * @return array
+     */
+    public function errorForCountryBudgetItem(array $activity): array
+    {
+        $countryBudgetItems = Arr::get($activity, 'country_budget_items', []);
+
+        return $this->getBaseRules((new CountryBudgetItemRequest())->getErrorsForCountryBudgetItem($countryBudgetItems), 'country_budget_items', $countryBudgetItems, false);
+    }
+
+    /**
+     * returns rules for HumanitarianScope.
+     *
+     * @param array $activity
+     *
+     * @return array
+     */
+    public function errorForHumanitarianScope(array $activity): array
+    {
+        return (new HumanitarianScopeRequest())->getErrorsForHumanitarianScope(Arr::get($activity, 'humanitarian_scope', []));
+    }
+
+    /**
+     * Get rules for Policy Marker.
+     *
+     * @param array $activity
+     * @return array
+     */
+    public function errorForPolicyMarker(array $activity): array
+    {
+        return (new PolicyMarkerRequest())->getErrorsForPolicyMarker(Arr::get($activity, 'policy_marker', []));
+    }
+
+    /**
+     * Get rules for Budget.
+     *
+     * @param array $activity
+     * @return array
+     */
+    protected function errorForBudget(array $activity): array
+    {
+        return (new BudgetRequest())->getErrorsForBudget(Arr::get($activity, 'budget', []));
+    }
+
+    /**
+     * @param array $activity
+     * @return array
+     */
+    protected function errorForPlannedDisbursement(array $activity): array
+    {
+        return (new PlannedDisbursementRequest())->getErrorsForPlannedDisbursement(Arr::get($activity, 'planned_disbursement', []));
+    }
+
+    /**
+     * @param array $activity
+     * @return array
```

Comment:
- add a new line in between these two param and return 
- function description missing